blogposts/2020/performance-improvements-in-precise-code-intel.md

```diff
@@ -0,0 +1,283 @@
+---
+title: "Performance improvements in precise code intel"
+description: This post outlines many of the higher-level changes that were made in a direct effort to increase the performance of precise code intel queries, increase the performance of raw LSIF upload processing, and decrease the size of precise code intel bundles on disk.
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2020-06-17T10:00-08:00
+tags: [blog]
+slug: performance-improvements-in-precise-code-intel
+heroImage: /blog/flying-brain.png
+published: true
+---
+
+In [Sourcegraph 3.16](/blog/sourcegraph-3.16#performance-improvements-for-precise-code-intelligence), we mentioned a rewrite of the services composing the precise-code-intel backend from TypeScript to Go. There were many reasons for this rewrite, but there's one particular reason that I'd like to explore here: _We as a team know how to improve Go code that operates on large-scale data and we have less experience doing the same for TypeScript on the server side_.
+
+This is **not** to say anything negative about TypeScript - the language is a surprising joy to work in, I'll admit after nearly a year. This is **not** to say that it's impossible to write code that performs well in a Node.js environment - V8 is a true beast of engineering. This is **not** to say that there is a lack of tools in order to help developers profile and improve their code.
+
+This **is** to say that I think it was good move to perform the rewrite to allow the backend team to play to our strengths. Rewriting this code in a language for which we have a better mental model of semantics and performance, a better grasp of the ecosystem, and actual _actual_ experience writing high-performance code allows us to move with enough velocity in the future that the time spent rewriting will be paid off in short order. Opened the code up to all other backend developers at Sourcegraph where Go is a core competency is obviously a move in the correct direction.
```

Comment:
The last sentence in this paragraph seems a bit out of place?
```suggestion
This **is** to say that I think it was good move to perform the rewrite to allow the backend team to play to our strengths. Rewriting this code in a language for which we have a better mental model of semantics and performance, a better grasp of the ecosystem, and actual _actual_ experience writing high-performance code allows us to move with enough velocity in the future that the time spent rewriting will be paid off in short order. And opening up the code to all other backend developers at Sourcegraph where Go is a core competency is obviously a move in the correct direction.
```

Reply:
Did you mean to say actual *twice*?

Reply:
Nope, fixed.